src/consensus/params.h

```diff
@@ -141,6 +141,63 @@ class FundingStream
     FundingStreamAddress RecipientAddress(const Params& params, int nHeight) const;
 };
 
+enum ConsensusFeature : uint32_t {
+    // Index value for the maximum consensus feature ID.
+    MAX_FEATURES
+};
+const auto FIRST_CONSENSUS_FEATURE = MAX_FEATURES;
+
+template <class Feature>
+struct FeatureInfo {
+    std::vector<Feature> dependsOn;
```

Comment:
The `Feature` type is only ever instantiated with the `ConsensusFeature` enum in this PR. Either document why this flexibility is present (especially given that with this type construction `FeatureInfo` can only depend on other features with the same type), or remove the templating and just use `ConsensusFeature` directly.